From ccf6f95d6902fdb62e9f7f2ef5c50909438111bb Mon Sep 17 00:00:00 2001
From: ci7lus <7887955+ci7lus@users.noreply.github.com>
Date: Wed, 11 Feb 2026 13:17:04 +0900
Subject: [PATCH] refactor: Identify MPEG-TS demuxer in player timer using a
 new `is-mpegts` variable instead of `psz_name` string comparison.

---
 src/player/timer.c | 25 ++++++++-----------------
 1 file changed, 8 insertions(+), 17 deletions(-)

diff --git a/src/player/timer.c b/src/player/timer.c
index 5751bf10dd..bd9ba69692 100644
--- a/src/player/timer.c
+++ b/src/player/timer.c
@@ -373,26 +373,17 @@ vlc_player_UpdateTimerSource(vlc_player_t *player,
         if (priv->master && priv->master->p_demux)
         {
             demux_t *p_demux = priv->master->p_demux;
-            if (p_demux->psz_name)
+            /* TSデマクサが（識別用に）作成した定義済みの変数を確認 */
+            if (var_Type(p_demux, "ts-standard") != 0)
             {
-                if (strcmp(p_demux->psz_name, "ts") == 0)
-                {
-                    bool stream_can_fastseek = false;
-                    if (p_demux->s)
-                        vlc_stream_Control(p_demux->s, STREAM_CAN_FASTSEEK, &stream_can_fastseek);
+                bool stream_can_fastseek = false;
+                if (p_demux->s)
+                    vlc_stream_Control(p_demux->s, STREAM_CAN_FASTSEEK, &stream_can_fastseek);

-                    msg_Info(player, "TIMER: mpegts detected. stream_can_fastseek: %d", stream_can_fastseek);
-                    if (!stream_can_fastseek)
-                        trust_demux_pos = true;
-                }
-                else
-                {
-                    /* psz_name が異なる場合の確認用ログ */
-                    msg_Info(player, "TIMER: demuxer is not ts, but: %s", p_demux->psz_name);
-                }
+                msg_Dbg(player, "TIMER: MPEG-TS detected via ts-standard. stream_can_fastseek: %d", stream_can_fastseek);
+                if (!stream_can_fastseek)
+                    trust_demux_pos = true;
             }
-            else
-                msg_Info(player, "TIMER: p_demux->psz_name is NULL");
         }
         else
             msg_Info(player, "TIMER: priv->master or p_demux is NULL");
--
2.39.1

